src/ln/channelmanager.rs

```diff
@@ -120,6 +120,32 @@ enum PendingOutboundHTLC {
 	}
 }
 
+struct MsgHandleErrInternal {
+	err: msgs::HandleError,
+	needs_channel_force_close: bool,
+}
+impl MsgHandleErrInternal {
```

Comment:
Having a builder is great to simplify errors filling but I think it's force us to specify a method for each case as a71abac shows it.

Reply:
Hmm, maybe, though I also don't want to have ErrBuilder::new().msg().action().needs_close() every time we want to return an Err cause then we spend all our charachters on a line on error handling :/. Not sure if there's a great solution here, really, but I'm open to patches to play around with it.

Reply:
Alright, well nothing stopping us from cleaning up things with macros or flattening errors in a future PR.